ui/src/app/pages/admin/admin-users.tsx

```diff
@@ -133,50 +140,176 @@ export const AdminUsers = withUserProfile()(class extends React.Component<
     return new Date(date).toString().split(' ').slice(1, 5).join(' ');
   }
 
-  convertProfilesToFields(profiles: Profile[]) {
-    return profiles.map(p => ({
-      ...p,
-      name: <a onClick={() => navigate(['admin', 'users', usernameWithoutDomain(p.username)])}>
-        {p.familyName + ', ' + p.givenName}
+  getAccessModuleString(user: AdminTableUser, key: string): (string|React.ReactElement) {
+    const completionTime = user[key + 'CompletionTime'];
+    const bypassTime = user[key + 'BypassTime'];
+
+    if (completionTime) {
+      const completionTimeString = moment(completionTime).format('lll');
+      return <TooltipTrigger content={`Completed at ${completionTimeString}`}>
+        <span>✔</span>
+      </TooltipTrigger>;
+    } else if (bypassTime) {
+      const bypassTimeString = moment(bypassTime).format('lll');
+      return <TooltipTrigger content={`Bypassed at ${bypassTimeString}`}>
+        <span>B</span>
+      </TooltipTrigger>;
+    } else {
+      return '';
+    }
+  }
+
+  convertProfilesToFields(users: AdminTableUser[]) {
+    return users.map(user => ({
+      audit: <a
+        href={`/admin/user-audit/${usernameWithoutDomain(user.username)}`}
+        target='_blank'>
+        link
+      </a>,
+      bypass: <AdminUserBypass user={{...user}}/>,
+      complianceTraining: this.getAccessModuleString(user, 'complianceTraining'),
+      contactEmail: user.contactEmail,
+      dataUseAgreement: this.getAccessModuleString(user, 'dataUseAgreement'),
+      eraCommons: this.getAccessModuleString(user, 'eraCommons'),
+      firstRegistrationCompletionTime: moment.unix(user.firstRegistrationCompletionTime / 1000).format('lll'),
```

Comment:
What happens when this value is not present?

Reply:
Good catch – fixed with some explicit empty string handling.